peps/pep-0763.rst

```diff
@@ -0,0 +1,275 @@
+PEP: 763
+Title: Limiting deletions on PyPI
+Author: William Woodruff <william@yossarian.net>,
+        Alexis Challande <alexis.challande@trailofbits.com>
+Sponsor: Donald Stufft <donald@stufft.io>
+PEP-Delegate: Donald Stufft <donald@stufft.io>
+Discussions-To: TODO
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 07-Oct-2024
+Post-History: `09-Jul-2022 <https://discuss.python.org/t/stop-allowing-deleting-things-from-pypi/17227>`__,
+              `01-Oct-2024 <https://discuss.python.org/t/pre-pep-limiting-deletions-on-pypi/66351>`__
+
+Abstract
+========
+
+This PEP proposes limiting user-level deletions of projects (including release
+and file-level deletions) on PyPI, in favor of the "yanking" mechanism
+adopted by PyPI with :pep:`592`.
+
+In particular, the PEP proposes a time-based deletion criterion via
+which projects (and their releases and files) are protected from *deletion*
+after an initial grace period of 72 hours following their creation.
+An exception to this criteria is made for versions and files that are
+marked with :ref:`pre-release specifiers <packaging:version-specifiers>`,
+which will remain deletable at all times.
+
+This PEP does not propose changes to administrator-initiated deletions, e.g.
+for moderation or security purposes.
+
+Rationale and Motivation
+========================
+
+As observed in :pep:`592`, user-level deletion of projects on PyPI
+enables a `Catch-22 <https://www.merriam-webster.com/dictionary/catch-22>`_
+of dependency breakage:
+
+    Whenever a project detects that a particular release on PyPI might be
+    broken, they oftentimes will want to prevent further users from
+    inadvertently using that version. However, the obvious solution of
+    deleting the existing file from a repository will break users who have
+    followed best practices and pinned to a specific version of the project.
+
+    This leaves projects in a catch-22 situation where new projects may be pulling
+    down this known broken version, but if they do anything to prevent that they’ll
+    break projects that are already using it.
```

Comment:
```suggestion
    This leaves a project author in a catch-22 situation where other projects may be pulling
    down this known broken version, but if the project author does anything to prevent that, it
    breaks projects where the dependency has been pinned for use.
```

Reply:
Same as https://github.com/python/peps/pull/4080/files#r1817037637 -- I like this rephrasing but I don't know whether it's kosher to rewrite a direct quote 😅